docs/excel/custom-functions-parameter-options.md

```diff
@@ -0,0 +1,157 @@
+        secondHighest = values[i][j];
+      }
+    }
+  }
+  return secondHighest;
+}
+```
+
+## Invocation context parameter
+
+Every custom function automatically contains an `invocationContext` argument, which is invisibly present at the end of any list of parameters. Even if you declare no parameters, your custom function will have this parameter. This argument is not explicitly stated, so it does not appear to the end user in Excel, nor is it present in the code in your script files.
+
+In the following code sample, you'll see the invocationContext explicitly stated for your reference.
+
+```JavaScript
+subtract(one, two, invocationContext) {
+  return one - two;
+}
+```
+
+The parameter allows you to get the context of the invoking cell, which can be helpful in some scenarios including [discovering the address of a cell which invoke a custom function](#addressing-cells-context-parameter).
+
+## Addressing cell's context parameter
+
+In some cases you'll need to get the address of the cell that invoked your custom function. This may be useful in the following types of scenarios:
+
+- Formatting ranges: Use the cell's address as the key to store information in [AsyncStorage](https://docs.microsoft.com/office/dev/add-ins/excel/custom-functions-runtime#storing-and-accessing-data). Then, use [onCalculated](https://docs.microsoft.com/javascript/api/excel/excel.worksheet#oncalculated) in Excel to load the key from `AsyncStorage`.
+- Displaying cached values: If your function is used offline, display stored cached values from `AsyncStorage` using `onCalculated`.
+- Reconciliation: Use the cell's address to discover an origin cell to help you reconcile where processing is occurring.
+
+The information about a cell's address is exposed only if `requiresAddress` is marked as `true` in the function's JSON metadata file. The following sample gives an example of this:
+
+```JSON
+{
+   "id": "ADDTIME",
+   "name": "ADDTIME",
+   "description": "Display current date and add the amount of hours to it designated by the parameter",
+   "helpUrl": "http://www.contoso.com",
+   "result": {
+      "type": "number",
+      "dimensionality": "scalar"
+   },
+   "parameters": [
+      {
+         "name": "Additional time",
+         "description": "Amount of hours to increase current date by",
+         "type": "number",
+         "dimensionality": "scalar"
+      }
+   ],
+   "options": {
+      "requiresAddress": true
+   }
+}
+```
+
+In the script file (**./src/customfunctions.js** or **./src/customfunctions.ts**), you'll also need to add a `getAddress` function to find a cell's address. This function may take parameters, as shown in the following sample as `parameter1`. The last parameter will always be `invocationContext`, an object containing the cell's location that Excel passes down when `requiresAddress` is marked as `true` in your JSON metadata file.
+
+```js
+function getAddress(parameter1, invocationContext) {
```

Comment:
Is this function explicitly necessary? I'm a little confused, since this seems like a helper function, but that doesn't seem necessary if all functions get the invocationContext and we've ensured it has an address through the JSON.

Reply:
I'm also confused if this is a helper function or a custom function.